files/enc.rb

```diff
@@ -35,9 +35,33 @@ def puppetuser
   SETTINGS[:puppetuser] || 'puppet'
 end
 
-def stat_file(certname)
-  FileUtils.mkdir_p "#{puppetdir}/yaml/foreman/"
-  "#{puppetdir}/yaml/foreman/#{certname}.yaml"
+def fact_extension
+  SETTINGS[:fact_extension] || 'yaml'
+end
+
+def fact_directory
+  if fact_extension == 'yaml'
+    File.join(puppetdir, fact_extension, 'facts')
+  else
+    File.join(puppetdir, 'server_data', 'facts')
+  end
+end
+
+def fact_file(certname)
+  File.join(fact_directory, "#{certname}.#{fact_extension}")
+end
+
+def fact_files
+  Dir[File.join(fact_directory, "*.#{fact_extension}")]
+end
+
+def certname_from_filename(filename)
+  File.basename(filename, ".#{fact_extension}")
+end
+
+def stat_file(certname, extension='yaml')
+  FileUtils.mkdir_p "#{puppetdir}/#{extension}/foreman/"
```

Comment:
I take it FileUtils in this context does not need to be explicitly required?

Reply:
No, that appears to always be loaded. It was already there previously so it's been working for a long time.